src/store/actions.js

```diff
@@ -2,22 +2,83 @@ import api from '@molgenis/molgenis-api-client'
 import EntityToTreeMapper from '../util/EntityToTreeMapper'
 import {
   SET_TREE_DATA,
-  SET_ERROR
+  SET_ERROR, SET_CORE_VARIABLE_COLUMNS, SET_CORE_VARIABLE_DATA, SET_COHORT_DATA, SET_HARMONIZATION_DATA,
+  SET_RAW_TREE_DATA, SET_SOURCE_VARIABLES
 } from './mutations'
+import _ from 'lodash'
+import EntityToCoreVariableMapper from '../util/EntityToCoreVariableMapper'
 
 /* ACTION CONSTANTS */
 export const GET_TREE_DATA = '__GET_TREE_DATA__'
+export const GET_CORE_VARIABLES = '__GET_CORE_VARIABLE_DATA__'
+export const GET_COHORTS = '__GET_COHORT_DATA__'
+export const GET_HARMONIZATIONS = '__GET_HARMONIZATIONS__'
+export const GET_SOURCE_VARIABLES = '__GET_SOURCE_VARIABLES__'
 
 /* API PATHS */
 const TREE_API_PATH = '/api/v2/UI_Menu'
+const CORE_VARIABLE_API_PATH = '/api/v2/LifeCycle_CoreVariables'
+const COHORT_API_PATH = '/api/v2/LifeCycle_Cohorts'
+const HARMONIZATION_API_PATH = '/api/v2/LifeCycle_Harmonizations'
+const SOURCE_VARIABLES_API_PATH = '/api/v2/LifeCycle_SourceVariables'
 
 export default {
   [GET_TREE_DATA] ({state, commit}) {
     api.get(TREE_API_PATH).then(response => {
-      const data = EntityToTreeMapper.generateTreeData(response, state.tree.settings)
+      commit(SET_RAW_TREE_DATA, response.items)
+      // We do this because we can test this method more easily
+      // We used to use the getters in the generateTreeData method
+      const clonedResponse = _.cloneDeep(response)
+      const data = EntityToTreeMapper.generateTreeData(clonedResponse.items, state.tree.settings)
       commit(SET_TREE_DATA, data)
     }, error => {
       commit(SET_ERROR, error)
     })
+  },
+  [GET_CORE_VARIABLES] ({state, commit, getters}, treeId) {
+    const treeLeaf = getters.getRawTreeData.find((item) => {
+      return item.key === treeId
+    })
+    const variables = Object.keys(treeLeaf.variables).map(function (variable) {
```

Comment:
Logic like this is a typical getter, you use rawTreeData to retrieve a leaf, then get the variables from that leaf.
Seeing as you duplicate this code in a later action as well, I would suggest writing a getter or a util function.

Makes the actions easier to test as well ;)

Reply:
Will be refactored in getters-test PR